src/VisualStudio/Core/Def/Implementation/KeybindingReset/KeybindingResetDetector.cs

```diff
@@ -207,19 +219,16 @@ private async Task UpdateStateMachineWorkerAsync(CancellationToken cancellationT
                     if (currentStatus != ReSharperStatus.Enabled)
                     {
                         // E->N or E->S. Set NeedsReset. Pop the gold bar to the user.
-                        options = options.WithChangedOption(KeybindingResetOptions.NeedsReset, true);
+                        needsReset = true;
                     }
 
                     // Else is E->E. No actions to take
                     break;
             }
 
-            // Apply the new options.
-            // We need to switch to UI thread to invoke TryApplyChanges.
-            await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
-            _workspace.TryApplyChanges(_workspace.CurrentSolution.WithOptions(options));
+            _optionService.SetGlobalOptions(s_statusOptions, ImmutableArray.Create<object>(currentStatus, needsReset));
```

Comment:
I don't know if we really need to, but it seems reasonable since the code implements a state machine that transitions between state that's defined by both values.

Reply:
I'm not sure what you mean by reordering something and breaking it.